src/guides/v2.3/graphql/mutations/generate-customer-token.md

```diff
@@ -57,6 +57,14 @@ Attribute |  Data Type | Description
 --- | --- | ---
 `token` | String | The customer token
 
+## Errors
+
+Error | Description
+--- | ---
+`Specify the "email" value.` | Specified value in the `email` attribute is empty.
```

Comment:
Since email and password are marked as required in the GraphQl schema, aren't they supposed to be validated on the schema processor's (Webonyx) end? 

[Mutation Declaration](https://github.com/magento/magento2/blob/69b70b908ad7cb5b32df53ce1a00328897270487/app/code/Magento/CustomerGraphQl/etc/schema.graphqls#L12)

Reply:
Hi, @rogyar !

> Since email and password are marked as required in the GraphQl schema, aren't they supposed to be validated on the schema processor's (Webonyx) end?
> 
> [Mutation Declaration](https://github.com/magento/magento2/blob/69b70b908ad7cb5b32df53ce1a00328897270487/app/code/Magento/CustomerGraphQl/etc/schema.graphqls#L12)

Good notice.

However, `Webonix` covers cases when email or password are not set. For example,

<img width="775" alt="6447" src="https://user-images.githubusercontent.com/13585327/73053672-03041280-3e91-11ea-811e-0e61f1a82646.png">

When email has an empty value then `Webonix` passes this value and moves it to resolver - `Magento\CustomerGraphQl\Model\Resolver\GenerateCustomerToken`:

<img width="788" alt="6447_01" src="https://user-images.githubusercontent.com/13585327/73053776-51b1ac80-3e91-11ea-9d42-5bd03e0eb6e1.png">

So, all is ok here.

Thank you!

Reply:
That's really good to know, thank you.